staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -118,14 +130,44 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, namespace, na
 	var patchType types.PatchType
 	var patch []byte
 
-	if p.OpenapiSchema != nil {
-		// if openapischema is used, we'll try to get required patch type for this GVK from Open API.
-		// if it fails or could not find any patch type, fall back to baked-in patch type determination.
-		if patchType, err = p.getPatchTypeFromOpenAPI(p.Mapping.GroupVersionKind); err == nil && patchType == types.StrategicMergePatchType {
-			patch, err = p.buildStrategicMergeFromOpenAPI(original, modified, current)
-			if err != nil {
-				// Warn user about problem and continue strategic merge patching using builtin types.
-				fmt.Fprintf(errOut, "warning: error calculating patch from openapi spec: %v\n", err)
+	if p.OpenAPIV3Root != nil {
+		gvkSupported, err := p.gvkSupportsPatchOpenAPIV3(p.Mapping.GroupVersionKind)
+		if err != nil {
+			// Realistically this error logging is not needed (not present in V2),
+			// but would help us in debugging if users encounter a problem
+			// with OpenAPI V3 not present in V2.
+			klog.V(5).Infof("warning: OpenAPI V3 path does not exist - group: %s, version %s, kind %s\n",
+				p.Mapping.GroupVersionKind.Group, p.Mapping.GroupVersionKind.Version, p.Mapping.GroupVersionKind.Kind)
+		} else {
+			if gvkSupported {
+				patch, err = p.buildStrategicMergePatchFromOpenAPIV3(original, modified, current)
+				if err != nil {
+					// Fall back to OpenAPI V2 if there is a problem
+					// We should remove the fallback in the future,
+					// but for the first release it might be beneficial
+					// to fall back to OpenAPI V2 while logging the error
+					// and seeing if we get any bug reports.
+					fmt.Fprintf(errOut, "warning: error calculating patch from openapi v3 spec: %v\n", err)
+				} else {
+					patchType = types.StrategicMergePatchType
+				}
+			} else {
+				klog.V(5).Infof("warning: OpenAPI V3 path does not support strategic merge patch - group: %s, version %s, kind %s\n",
```

Comment:
Should we really have a warning about this? This just means it's a CRD?

Reply:
We shouldn't warn, hence the v5 info log. I'd like to make sure every case is logged in case these rare cases do come up and users can provide us with the full log for better debugging. lmk if you think this should be a different log level or removed completely.

Reply:
Yeah, if that's super mild then that's fine thanks.